File: Backend/IronMan.Core/Dtos/Authentication/RegisterRequestDto.cs
using IronMan.Data.Entities;
using System.ComponentModel.DataAnnotations;

namespace IronMan.Core.Dtos.Authentication
{
    public class RegisterRequestDto
    {
        [Required]
        public string Title { get; set; }
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        [Required]
        [MinLength(6)]
        public string Password { get; set; }
        [Required]
        [Compare("Password")]
        public string ConfirmPassword { get; set; }
        [Required]
        public Role Role { get; set; }
        [Required]
        public int Department { get; set; }
        [Required]
        public int Location { get; set; }

        [Range(typeof(bool), "true", "true")]
        public bool AcceptTerms { get; set; }
    }
}

File: Frontend/src/proxy/Interfaces/Authentication/register-request-dto.ts
import { Roles } from "src/proxy/Enums/roles";

export interface RegisterRequestDto {
    title: string;
    firstName: string;
    lastName: string;
    email: string;
    password: string;
    confirmPassword: string;
    role: Roles;
    department: number;
    acceptTerms: boolean;
}
